plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -918,3 +918,41 @@ func (checker *PodAffinityChecker) NodeMatchPodAffinityAntiAffinity(pod *api.Pod
 	}
 	return true
 }
+
+// Determine if a pod is scheduled with best-effort QoS
+// Definition of best-effort pod by vishh: https://github.com/kubernetes/kubernetes/pull/14943
+func isPodBestEffort(pod *api.Pod) bool {
+	// A Pod is BestEffort iff requests & limits are not specified
+	// for any resource across all containers.
+	for _, container := range pod.Spec.Containers {
+		resources := container.Resources
+		if resources.Limits != nil || resources.Requests != nil {
+			return false
+		}
+	}
+
+	return true
+}
+
+// MemoryPressurePredicate checks if a node is under memory pressure
+// and a pod is schedulable on it.
+func NodeMemoryPressurePredicate(pod *api.Pod, nodeInfo *schedulercache.NodeInfo) (bool, error) {
```

Comment:
I wanted to be consistent with the naming style:

* ``NewPodAffinityPredicate``
* ``GeneralPredicates``
* ``NewMaxPDVolumeCountPredicate``
* ``NewVolumeZonePredicate``
* ``NewNodeLabelPredicate``
* ``NewServiceAffinityPredicate``
* ...

Choose the name by predicate category (its purpose), not by what it does. Though, all the predicate names in the list are actually factories.

``NodeMemoryPressurePredicate`` by its name covers all checks that has something to do with memory pressure state. Though, ``NodeNotUnderMemoryPressure`` tells exactly that the predicate checks if the node is not under memory pressure. In future, it could make sense to check if the node is under "almost memory pressure" or "getting close to memory pressure".

What about ``CheckNodeMemoryPressurePredicate``? This says it checks the condition but still keeps it general enough to make the predicate extendible for more "memory pressure" type conditions.